### PR TITLE
Use auth back as admin control and add new structure to manage gulp w…

### PR DESCRIPTION
**Pull request by gbalabasquer**
…hitelisting

+ option to whitelist every address (not sure if it is the best approach).

### CODE REVIEW

**Review by iamchrissmith — APPROVED** [review]
Overall, I like this. While I think the `address(0)` for global whitelisting is a bit "hidden", I don't know that adding a separate on/off toggle for that would really be a better implementation since its not really a public interface. 

I was wondering if we should automatically `hope` a `rely`ed address. But this is probably fine and we can just make two calls for addresses that should be both.

File: src/RestrictedTokenFaucet.sol [hunk 1]
```diff
@@ -8,52 +8,60 @@ interface ERC20Like {
...
+    modifier auth {
+        require(wards[msg.sender] == 1, "token-faucet/no-auth");
+        _;
+    }
+    // --- Gulp Whitelist ---
+    mapping (address => uint) public list;
+    function hope(address guy) public auth { list[guy] = 1; }
+    function nope(address guy) public auth { list[guy] = 0; }
+
     address public owner;
```
Comment (iamchrissmith) [inline src/RestrictedTokenFaucet.sol]: We don't need this anymore

**Review by iamchrissmith — CHANGES_REQUESTED** [review]

File: src/RestrictedTokenFaucet.sol (hunk 1 above)
Comment (gbalabasquer) [reply src/RestrictedTokenFaucet.sol]: what do you mean here?

Comment (iamchrissmith) [reply src/RestrictedTokenFaucet.sol]: we don't need the `address public owner` storage anymore with `auth` and `hope`/`nope`

Comment (gbalabasquer) [reply src/RestrictedTokenFaucet.sol]: Ahh ok, will remove it now.

**Comment by gbalabasquer** [comment]
@iamchrissmith Are we ready to get this merged to `restricted-token-faucet` then to `master`?

**Review by iamchrissmith — APPROVED** [review]

**Comment by iamchrissmith** [comment]
LGTM

**Pull request merged**